crates/codegen/spec/src/model.rs

```diff
@@ -91,6 +91,17 @@ impl SpecTopic {
 
         for item in &topic.items {
             items.push(item.name().to_owned());
+
+            // We need to also add any precedence expressions, as they define
+            // items but are not direct children of the topic
+            if let Item::Precedence {
+                item: precedence_item,
+            } = item
+            {
+                for precedence_expr in &precedence_item.as_ref().precedence_expressions {
+                    items.push(precedence_expr.as_ref().name.clone());
+                }
```

Comment:
Great catch!

nit suggestion: I believe `Rc<>` will `deref()` automatically, so need to call `as_ref()` here:

```suggestion
                for precedence_expr in &precedence_item.precedence_expressions {
                    items.push(precedence_expr.name.clone());
                }
```